themes/startwords/layouts/shortcodes/figure.html

```diff
@@ -1,14 +1,46 @@
+{{/* get page resource matching filename specified as "src" in shortcode */}}
+{{ $img := .Page.Resources.GetMatch (printf "*%s*" (.Get "src")) }}
+
+{{/* set image derivative sizes */}}
+{{ $tinyw := default "500x" }}
+{{ $smallw := default "800x" }}
+{{ $mediumw := default "1200x" }}
+{{ $largew := default "1500x" }}
+
+{{/* resize the image to the given sizes */}}
+{{ .Scratch.Set "tiny" ($img.Resize $tinyw) }}
+{{ .Scratch.Set "small" ($img.Resize $smallw) }}
+{{ .Scratch.Set "medium" ($img.Resize $mediumw) }}
+{{ .Scratch.Set "large" ($img.Resize $largew) }}
+
+{{/* store processed images for output in srcset */}}
+{{ $tiny := .Scratch.Get "tiny" }}
+{{ $small := .Scratch.Get "small" }}
+{{ $medium := .Scratch.Get "medium" }}
+{{ $large := .Scratch.Get "large" }}
+
 <figure{{ with .Get "class" }} class="{{ . }}"{{ end }}{{ if .Get "extdesc_id" }} aria-describedby="{{ .Get "desc_id" }}"{{ end }}>
     {{- if .Get "link" -}}
         <a href="{{ .Get "link" }}"{{ with .Get "target" }} target="{{ . }}"{{ end }}{{ with .Get "rel" }} rel="{{ . }}"{{ end }}>
     {{- end }}
-    <img src="{{ .Get "src" }}" role="img"
+    <img src="{{ $img.RelPermalink }}" role="img"
```

Comment:
Should `src` use `large` rather than the original image? Since that's the biggest version we want to display, and the original could be larger than we actually want to include in the website.

Reply:
oh, also — what do you think about adding `loading=lazy` while you're here?

Reply:
good call on `loading=lazy`, I keep forgetting that's a thing! I think for using `large` it just depends on if we expect the majority of images to be larger than 1500px or not...I guess it's probably better to serve a larger image that gets scaled than one that's pixelated or illegible anyway, so I'll change it.